### PR TITLE
sdl_ttf: add version 2.0.18

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **sdl_ttf/2.0.18**

- patch sources only for sdl_ttf/2.0.16, because CMakeLists.txt is already fixed.
- modify url of source because libsdl.org's 2.0.18 sources doesn't have CMakeLists.txt(I don't know the reason)

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/sdl_ttf/all/conanfile.py [hunk 1]
```diff
@@ -88,9 +88,14 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "SDL2_ttf.framework"))
 
     def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "SDL2_ttf")
+        self.cpp_info.set_property("cmake_target_name", "SDL2_ttf::SDL2_ttf")
+        self.cpp_info.set_property("pkg_config_name", "SDL2_ttf")
```
Comment (SpaceIm) [inline recipes/sdl_ttf/all/conanfile.py:93]: bump `required_conan_version` to 1.43.0 please

File: recipes/sdl_ttf/all/conanfile.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ def source(self):
 
     def _patch_sources(self):
         # Handle sdl2 static target
-        if not self.options["sdl"].shared:
+        if tools.Version(self.version) < "2.0.18" and not self.options["sdl"].shared:
             tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
                                   "SDL2::SDL2", "SDL2::SDL2-static")
```
Comment (SpaceIm) [inline recipes/sdl_ttf/all/conanfile.py:65]: No, it's even worse in 2.0.18: https://github.com/libsdl-org/SDL_ttf/blob/release-2.0.18/CMakeLists.txt#L68-L72

They link to static SDL2 when SDL2_ttf is static, and SDL2 shared when SDL2_ttf shared. We want all to enable all combinations.

File: recipes/sdl_ttf/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/sdl_ttf/all/conanfile.py:93]: Thank you. I fixed it.

File: recipes/sdl_ttf/all/conanfile.py (hunk 2 above)
Comment (toge) [reply recipes/sdl_ttf/all/conanfile.py:65]: Thank you for information.
I will fix this problem.

File: recipes/sdl_ttf/all/conanfile.py [hunk 3]
```diff
@@ -59,10 +63,9 @@ def source(self):
                   destination=self._source_subfolder, strip_root=True)
 
     def _patch_sources(self):
-        # Handle sdl2 static target
-        if not self.options["sdl"].shared:
-            tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
-                                  "SDL2::SDL2", "SDL2::SDL2-static")
```
Comment (SpaceIm) [inline recipes/sdl_ttf/all/conanfile.py:65]: Don't forget the patch for 2.0.15 as well because you have removed this block.

Comment (toge) [reply recipes/sdl_ttf/all/conanfile.py:65]: Sorry. I forget it.
I create patch for 2.0.15.

File: recipes/sdl_ttf/all/patches/cmake-fix-link-target-2.0.15.patch [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+ add_library(SDL2_ttf SDL_ttf.c SDL_ttf.h)
++
++if(TARGET SDL2::SDL2)
++  target_link_libraries(SDL2_ttf SDL2::SDL2 Freetype::Freetype)
++else ()
++  target_link_libraries(SDL2_ttf SDL2::SDL2-static Freetype::Freetype)
++endif ()
++
++
+ target_link_libraries(SDL2_ttf SDL2::SDL2 Freetype::Freetype)
```
Comment (SpaceIm) [inline recipes/sdl_ttf/all/patches/cmake-fix-link-target-2.0.15.patch:17]: This patch is not correct

Comment (toge) [reply recipes/sdl_ttf/all/patches/cmake-fix-link-target-2.0.15.patch:17]: Sorry for my misunderstanding.

Is following condition is right in 2.0.15?
```
if (BUILD_SHARED_LIBS)
```

Comment (SpaceIm) [reply recipes/sdl_ttf/all/patches/cmake-fix-link-target-2.0.15.patch:17]: `target_link_libraries(SDL2_ttf SDL2::SDL2 Freetype::Freetype)` must be removed and replaced by the block you have added.

Comment (toge) [reply recipes/sdl_ttf/all/patches/cmake-fix-link-target-2.0.15.patch:17]: Oh. I miss line 17.
Thank you!

**Review by SpaceIm — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**